src/deadline/nuke_adaptor/copycat_adaptor.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import json
+import logging
+import pathlib
+import re
+import subprocess
+import sys
+import threading
+import time
+from typing import Dict, List, Optional
+
+error_regex = re.compile(r".*ERROR: (.+)")
+step_complete_regex = re.compile(r"\[Step:(\d+)/(\d+)\].*")
+
+_current_percent_done = 0
+
+
+def _report_progress(current_step, total_steps):
```

Comment:
can you add a docstring for the method to help the other folks understand the code?